discovery-infra/tests/test_install.py

```diff
@@ -9,7 +14,8 @@ class TestInstall(BaseTest):
 
     @JunitTestSuite()
     @pytest.mark.parametrize("openshift_version", get_available_openshift_versions())
-    def test_install(self, nodes, cluster, openshift_version):
-        new_cluster = cluster(openshift_version=openshift_version)
-        new_cluster.prepare_for_install(nodes=nodes)
-        new_cluster.start_install_and_wait_for_installed()
+    def test_install(self, nodes: Nodes, cluster, openshift_version):
+        config = DefaultClusterConfig(openshift_version=openshift_version)
```

Comment:
I think that usage of DefaultClusterConfig is a little bit strange

Reply:
I think that if we already set all the cluster params we should not need to provide same parameters twice.

Reply:
maybe we can allow to override them but not to provide full config for sure

Reply:
I wanted to preserve the current flow and not pass all argument on Cluster initialization.
Parameters that were passed on Cluster initialization stays there and parameters that were passed on installation passed there.

Reply:
>I think that if we already set all the cluster params we should not need to provide same parameters twice.
This is the point where it passes the cluster parameters.

Reply:
not sure i understand why we need it in case we will have config class. What it will give us? It will require from user to create 2 classes that describes same cluster. Previously maybe there was a point for it (arguable though) but in case we deal with configuration class i think there is no point to preserve same logic. Still i think it will be nice to have an option to override those params.

Reply:
If there is no any case that we want to pass different parameters on installation (maybe we don't know all of theme at cluster initialization).
That mean that `prepare_for_install` will receive only nodes and will not be able to receive any other arguments (due to single config object that passes to Cluster on initialization)